crates/sui-types/src/storage.rs

```diff
@@ -54,3 +54,31 @@ impl<S: BackingPackageStore> BackingPackageStore for &S {
         BackingPackageStore::get_package(*self, package_id)
     }
 }
+
+impl<S: BackingPackageStore> BackingPackageStore for &mut S {
+    fn get_package(&self, package_id: &ObjectID) -> SuiResult<Option<Object>> {
+        BackingPackageStore::get_package(*self, package_id)
+    }
+}
+
+pub trait ParentSync {
+    fn get_latest_parent_entry_ref(&self, object_id: ObjectID) -> SuiResult<Option<ObjectRef>>;
+}
+
+impl<S: ParentSync> ParentSync for std::sync::Arc<S> {
+    fn get_latest_parent_entry_ref(&self, object_id: ObjectID) -> SuiResult<Option<ObjectRef>> {
+        ParentSync::get_latest_parent_entry_ref(self.as_ref(), object_id)
+    }
+}
+
+impl<S: ParentSync> ParentSync for &S {
+    fn get_latest_parent_entry_ref(&self, object_id: ObjectID) -> SuiResult<Option<ObjectRef>> {
+        ParentSync::get_latest_parent_entry_ref(*self, object_id)
+    }
+}
+
+impl<S: ParentSync> ParentSync for &mut S {
+    fn get_latest_parent_entry_ref(&self, object_id: ObjectID) -> SuiResult<Option<ObjectRef>> {
+        ParentSync::get_latest_parent_entry_ref(*self, object_id)
+    }
+}
```

Comment:
Do we really need these? I am under the impression that Rust can auto-deref for Arc and &

Reply:
I assumed we did because they were there for `BackingStore`, I can try removing them